source

```
@@ -79482,50 +79482,22 @@ callback <dfn>FrameRequestCallback</dfn> = void (<span>DOMHighResTimeStamp</span
 
   </dl>
 
-  <div w-nodev>
-
-  <p>The <dfn><code data-x="dom-document-domain">domain</code></dfn> attribute's getter must run
-  these steps:</p>
-
-  <ol>
-   <li><p>If this <code>Document</code> object does not have a <span
-   data-x="concept-document-bc">browsing context</span>, then return the empty string.
-
-   <li><p>Let <var>effectiveDomain</var> be this <code>Document</code>'s <span>origin</span>'s <span
-   data-x="concept-origin-effective-domain">effective domain</span>.
-
-   <li><p>If <var>effectiveDomain</var> is an <span data-x="concept-origin-opaque">opaque
-   origin</span>, then return the empty string.</p></li>
-
-   <li><p>Return <var>effectiveDomain</var>, <span data-x="host
-   serializer">serialized</span>.</p></li>
-  </ol>
-
-  <p>The <code data-x="dom-document-domain">domain</code> attribute's setter must run these
+  <p>To <dfn data-export="">validate the relaxation of an origin</dfn> for an <span>origin</span>
+  <var>currentOrigin</var> and a string <var>inputDomain</var>, the user agent must run these
   steps:</p>
 
   <ol>
-   <li><p>If this <code>Document</code> object has no <span data-x="concept-document-bc">browsing
-   context</span>, then throw a <span>"<code>SecurityError</code>"</span>
-   <code>DOMException</code>.</p></li>
-
-   <li><p>If this <code>Document</code> object's <span>active sandboxing flag set</span> has its
-   <span>sandboxed <code data-x="dom-document-domain">document.domain</code> browsing context
-   flag</span> set, then throw a <span>"<code>SecurityError</code>"</span>
-   <code>DOMException</code>.</p></li>
-
-   <li><p>If the given value is the empty string, then throw a
+   <li><p>If <var>inputDomain</var> is the empty string, then throw a
    <span>"<code>SecurityError</code>"</span> <code>DOMException</code>.</p></li>
```

If we make this an abstract operation I don't think we should throw exceptions. Instead we should return failure and let the caller algorithm deal with the failure. That makes the algorithm more easily reusable in other contexts, that might not want to throw this exception.